src/Compute/Compute/VirtualMachine/Config/RemoveAzureVMDataDiskCommand.cs

```diff
@@ -53,20 +60,36 @@ public override void ExecuteCmdlet()
                     var disks = storageProfile.DataDisks.ToList();
                     var comp = StringComparison.OrdinalIgnoreCase;
 
-                    if (DataDiskNames == null)
-                    {
-                        disks.Clear();
+                    if (ForceDetach != true) {
```

Comment:
it's probably a better practice for code readability to reference parameters explicitly by **this**.ForceDetach. so not to be confused by any local variables inside functions.